src/coreclr/src/jit/earlyprop.cpp

```diff
@@ -321,26 +319,7 @@ GenTree* Compiler::optEarlyPropRewriteTree(GenTree* tree, LocalNumberToNullCheck
     }
     else
     {
-        assert(tree->OperIsIndir());
-        if (gtIsVtableRef(tree))
-        {
-            // Don't propagate type handles that are used as null checks, which are usually in
-            // form of
-            //      *  stmtExpr  void  (top level)
-            //      \--*  indir     int
-            //          \--*  lclVar    ref    V02 loc0
-            if (compCurStmt->GetRootNode() == tree)
-            {
-                return nullptr;
-            }
-
-            objectRefPtr = tree->AsIndir()->Addr();
```

Comment:
This tree looks like the old form of a null-check that was replaced by the GT_NULLCHECK node